pkg/registry/apis/dashboard/legacy/sql_dashboards_test.go

```diff
@@ -0,0 +1,110 @@
+package legacy
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/services/provisioning"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestScanRow(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close() // nolint:errcheck
+
+	pathToFile := "path/to/file"
+	provisioner := provisioning.NewProvisioningServiceMock(context.Background())
+	provisioner.GetDashboardProvisionerResolvedPathFunc = func(name string) string { return "provisioner" }
+	store := &dashboardSqlAccess{
+		namespacer:   func(_ int64) string { return "default" },
+		provisioning: provisioner,
+	}
+
+	columns := []string{"orgId", "dashboard_id", "name", "folder_uid", "deleted", "plugin_id", "origin_name", "origin_path", "origin_hash", "origin_ts", "created", "createdBy", "createdByID", "updated", "updatedBy", "updatedByID", "version", "message", "data"}
+	id := int64(100)
+	title := "Test Dashboard"
+	folderUID := "folder123"
+	timestamp := time.Now()
+	k8sTimestamp := v1.Time{Time: timestamp}
+	version := int64(2)
+	message := "updated message"
+	createdUser := "creator"
+	updatedUser := "updator"
+
+	t.Run("Should scan a valid row correctly", func(t *testing.T) {
+		rows := sqlmock.NewRows(columns).AddRow(1, id, title, folderUID, nil, "", "", "", "", 0, timestamp, createdUser, 0, timestamp, updatedUser, 0, version, message, []byte(`{"key": "value"}`))
+		mock.ExpectQuery("SELECT *").WillReturnRows(rows)
+		resultRows, err := mockDB.Query("SELECT *")
+		require.NoError(t, err)
+		defer resultRows.Close() // nolint:errcheck
+		resultRows.Next()
+
+		row, err := store.scanRow(resultRows, false)
+		require.NoError(t, err)
+		require.NotNil(t, row)
+		require.Equal(t, "Test Dashboard", row.Dash.Name)
+		require.Equal(t, version, row.RV) // rv should be the dashboard version
+		require.Equal(t, v0alpha1.Unstructured{
+			Object: map[string]interface{}{"key": "value"},
+		}, row.Dash.Spec)
+		require.Equal(t, "default", row.Dash.Namespace)
+		require.Equal(t, row.token, &continueToken{orgId: int64(1), id: id})
```

Comment:
nit: flip expected and actual parameters
```suggestion
		require.Equal(t, &continueToken{orgId: int64(1), id: id}, row.token)
```

Reply:
good catch, thank you!